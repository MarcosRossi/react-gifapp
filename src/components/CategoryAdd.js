import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CategoryAdd = function ({ setCategory }) {
  const [categoryName, setcategoryName] = useState('');
  const handleChange = (e) => {
    setcategoryName(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (categoryName.trim().length <= 2) return;

    setCategory((cats) => [categoryName, ...cats]);
    setcategoryName('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        value={categoryName}
        onChange={handleChange}
        type='text'
        placeholder='Insert a new  category name'
      />
    </form>
  );
};

CategoryAdd.propTypes = {
  setCategory: PropTypes.func.isRequired,
};
export default CategoryAdd;
