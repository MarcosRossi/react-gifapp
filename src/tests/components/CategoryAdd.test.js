import { shallow } from 'enzyme';
import React from 'react';
import CategoryAdd from '../../components/CategoryAdd';

describe('Tests in CategoryAdd.js', () => {
  const setCategory = jest.fn();
  let wrapper = shallow(<CategoryAdd setCategory={setCategory} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<CategoryAdd setCategory={setCategory} />);
  });

  test('should create the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change the input value', () => {
    const input = wrapper.find('input');
    const value = 'Hello word';
    input.simulate('change', { target: { value } });

    const pText = wrapper.find('p').text();
    expect(pText.trim()).toBe(value);
  });

  test('should not posted because the value is empty', () => {
    const preventDefault = jest.fn();
    wrapper.find('form').simulate('submit', { preventDefault });
    expect(setCategory).not.toHaveBeenCalled();
  });

  test('should call setCategory and clean the input', () => {
    const value = 'Hello word II';

    const input = wrapper
      .find('input')
      .simulate('change', { target: { value } });

    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });

    expect(setCategory).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop('value')).toBe('');
  });
});
