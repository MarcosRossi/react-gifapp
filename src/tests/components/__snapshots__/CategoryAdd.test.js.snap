// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in CategoryAdd.js should create the component 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Insert a new  category name"
    type="text"
    value=""
  />
</form>
`;
